src/app/client/src/app/modules/resource/components/resource/resource.component.ts

```diff
@@ -77,25 +88,37 @@ export class ResourceComponent implements OnInit {
    */
   constructor(pageSectionService: PageApiService, toasterService: ToasterService, private playerService: PlayerService,
     resourceService: ResourceService, config: ConfigService, private activatedRoute: ActivatedRoute, router: Router,
-  public utilService: UtilService, private cacheService: CacheService, private browserCacheTtlService: BrowserCacheTtlService) {
+  public utilService: UtilService, private cacheService: CacheService, private browserCacheTtlService: BrowserCacheTtlService,
+  userService: UserService) {
     this.pageSectionService = pageSectionService;
     this.toasterService = toasterService;
     this.resourceService = resourceService;
     this.config = config;
     this.router = router;
     this.router.onSameUrlNavigation = 'reload';
     this.sortingOptions = this.config.dropDownConfig.FILTER.RESOURCES.sortingOptions;
+    this.userService = userService;
   }
   /**
   * Subscribe to getPageData api.
   */
   populatePageData() {
     this.showLoader = true;
     this.noResult = false;
+    let softConstraints = {};
+    const filters = _.pickBy(this.filters, value => value.length > 0);
+      if (this.viewinBrowser && _.isEmpty(this.queryParams)) {
+      filters.board = _.get(this.framework, 'board') ;
+      filters.medium = _.get(this.framework, 'medium') ;
+      filters.subject = _.get(this.framework, 'subject');
+      filters.gradeLevel = _.get(this.framework, 'gradeLevel');
+      softConstraints = { subject: 98, medium: 99,  board: 100 };
+    }
     const option = {
       source: 'web',
       name: 'Resource',
-      filters: _.pickBy(this.filters, value => value.length > 0),
+      filters: filters,
+      softConstraints: softConstraints,
```

Comment:
This is ordering and sorting should be similar to explore page, contents with first board with index 0 or 1 should come first